src/rpc_request.rs

```diff
@@ -41,6 +42,18 @@ pub fn get_rpc_request_str(rpc_addr: SocketAddr) -> String {
     format!("http://{}", rpc_addr)
 }
 
+pub trait RpcRequestHandler {
+    fn make_rpc_request(
+        &self,
+        client: &RpcClient,
+        id: u64,
+        params: Option<Value>,
```

Comment:
How come you pass `RpcRequest` to `Rpu` and not here (just before `params`)?

Reply:
Oh yeah, that could work. Would presumably need to be passed into to all of the Rpu methods. I'll work that up.